python/samples/concepts/memory/azure_ai_search_hotel_sample.py

```diff
@@ -0,0 +1,158 @@
+from semantic_kernel.data import (
+    FilterClause,
+    VectorSearchOptions,
+    VectorStoreRecordDataField,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordUtils,
+    VectorStoreRecordVectorField,
+    vectorstoremodel,
+)
+
+
+@vectorstoremodel
+class HotelSampleClass(BaseModel):
+    hotel_id: Annotated[str, VectorStoreRecordKeyField]
+    hotel_name: Annotated[str | None, VectorStoreRecordDataField()] = None
+    description: Annotated[
+        str, VectorStoreRecordDataField(has_embedding=True, embedding_property_name="description_vector")
+    ]
+    description_vector: Annotated[
+        list[float] | None,
+        VectorStoreRecordVectorField(
+            dimensions=1536,
+            local_embedding=True,
+            embedding_settings={"embedding": OpenAIEmbeddingPromptExecutionSettings(dimensions=1536)},
+        ),
+    ] = None
+    description_fr: Annotated[
+        str, VectorStoreRecordDataField(has_embedding=True, embedding_property_name="description_fr_vector")
+    ]
+    description_fr_vector: Annotated[
+        list[float] | None,
+        VectorStoreRecordVectorField(
+            dimensions=1536,
+            local_embedding=True,
+            embedding_settings={"embedding": OpenAIEmbeddingPromptExecutionSettings(dimensions=1536)},
+        ),
+    ] = None
+    category: Annotated[str, VectorStoreRecordDataField()]
+    tags: Annotated[list[str], VectorStoreRecordDataField()]
+    parking_included: Annotated[bool | None, VectorStoreRecordDataField()] = None
+    last_renovation_date: Annotated[str | None, VectorStoreRecordDataField()] = None
+    rating: Annotated[float, VectorStoreRecordDataField()]
+    location: Annotated[dict[str, Any], VectorStoreRecordDataField()]
+    address: Annotated[dict[str, str | None], VectorStoreRecordDataField()]
+    rooms: Annotated[list[dict[str, Any]], VectorStoreRecordDataField()]
+
+
+HotelSampleClassType = TypeVar("HotelSampleClassType", bound=HotelSampleClass)
+
+
+async def add_vectors(store, vectorizer):
+    ids: list[str] = [res.get("hotel_id") async for res in await store.search_client.search(select="hotel_id")]
+    print("sample id:", ids[0])
+
+    hotels = await store.get_batch(ids)
+    if hotels is not None and isinstance(hotels, list):
+        for hotel in hotels:
+            if not hotel.description_vector or not hotel.description_fr_vector:
+                hotel = await vectorizer.add_vector_to_records(hotel, HotelSampleClass)
+                await store.upsert(hotel)
+
+
+async def main(first_run: bool = False):
+    kernel = Kernel()
+    embeddings = OpenAITextEmbedding(service_id="embedding", ai_model_id="text-embedding-3-small")
+    kernel.add_service(embeddings)
+    vectorizer = VectorStoreRecordUtils(kernel)
+    store = AzureAISearchCollection[HotelSampleClassType](
+        collection_name="hotels-sample-index", data_model_type=HotelSampleClass
+    )
+    if not await store.does_collection_exist():
```

Comment:
There is a pre-req to being able to work with this sample? Can we add a lot of comments about what steps the user should take so they don't run into this error?